### DISCUSSION TITLE
Question on Blocks/Inserts

### DISCUSSION THREAD

**Discussion by BigAloz** [haplokuon/netDxf#456] [Q&A]
I am trying to draw blocks (i.e. Inserts) I need to access the drawing primitives from the Block structure 

I am getting confused at how to do this.
I can access the Blok I require form
             Block aBlock = dxf.Blocks[dxfInsert.ElementAt(i).Block.Name];

However I can't seem to access the drawing primitive (line , text, circle etc).

I have tried this
                for(int k=0; k < aBlock.Entities.Count;k++)
                {
                    netDxf.Entities.EntityObject ts = aBlock.Entities.ElementAt(k);
                    if (ts.GetType() == netDxf.Entities.Line)
                        ;

Any suggestions appreciated. I only started looking at this library two days ago so am fumbling around in the dark abit.

**Comment by BigAloz** [comment]
Managed to work it out

               Block aBlock = dxf.Blocks[dxfInsert.ElementAt(i).Block.Name];
                foreach (EntityObject entity in aBlock.Entities)
                {
                        switch (entity.Type)
                        {
                            case EntityType.Circle:
.......etc

**Comment by DavidWishengrad** [comment]
Thank you. It will probably help me later.
